libdnf5-cli/output/transactioninfo.cpp

```diff
@@ -45,4 +48,28 @@ void print_transaction_info(libdnf5::transaction::Transaction & transaction) {
     print_transaction_item_table(transaction.get_comps_environments(), "Environments altered:");
 }
 
+std::string generate_user_info_str(uint32_t user_id) {
+    auto results = std::to_string(user_id);
+    auto user_info = getpwuid(user_id);
+
+    if (user_info == NULL) {
+        return results;
+    };
+
+    // If gecos information does exists, it is generally CSV formatted.
+    // This section assumes that the first entry in the gecos information is the display name.
+    if (user_info->pw_gecos != NULL && !std::string(user_info->pw_gecos).empty()) {
+        std::stringstream s_stream(user_info->pw_gecos);
+        std::string display_name;
+        std::getline(s_stream, display_name, ',');
+        results += std::format(" {}", display_name);
+    };
+
+    if (user_info->pw_name != NULL && !std::string(user_info->pw_name).empty()) {
```

Comment:
The condition `if (user_info->pw_name != NULL && !std::string(user_info->pw_name).empty())` could be replaced by `if (user_info->pw_name && strlen(user_info->pw_name) > 0)` or `if (user_info->pw_name != nullptr && strlen(user_info->pw_name) > 0)`.

It will not create any temporal object (std::string) to only check the length of `char *`